Recognizer.py

```diff
@@ -2,47 +2,54 @@
 # -- this code is licensed GPLv3
 # Copyright 2013 Jezra
 
-import pygst
-pygst.require('0.10')
-import gst
+import gi
+gi.require_version('Gst', '1.0')
 import os.path
-import gobject
+from gi.repository import GObject, Gst
+Gst.init(None)
 
 #define some global variables
 this_dir = os.path.dirname( os.path.abspath(__file__) )
 
 
-class Recognizer(gobject.GObject):
+class Recognizer(GObject.GObject):
 	__gsignals__ = {
-		'finished' : (gobject.SIGNAL_RUN_LAST, gobject.TYPE_NONE, (gobject.TYPE_STRING,))
+		'finished' : (GObject.SIGNAL_RUN_LAST, GObject.TYPE_NONE, (GObject.TYPE_STRING,))
 	}
 	def __init__(self, language_file, dictionary_file, src = None):
-		gobject.GObject.__init__(self)
+		GObject.GObject.__init__(self)
 		self.commands = {}
 		if src:
 			audio_src = 'alsasrc device="hw:%d,0"' % (src)
 		else:
 			audio_src = 'autoaudiosrc'
 
 		#build the pipeline
-		cmd = audio_src+' ! audioconvert ! audioresample ! vader name=vad ! pocketsphinx name=asr ! appsink sync=false'
-		self.pipeline=gst.parse_launch( cmd )
+		cmd = audio_src +' ! audioconvert ! audioresample ' + '! pocketsphinx name=asr ! appsink sync=false'
+		self.pipeline=Gst.parse_launch( cmd )
 		#get the Auto Speech Recognition piece
 		asr=self.pipeline.get_by_name('asr')
-		asr.connect('result', self.result)
+		bus=self.pipeline.get_bus()
+		bus.add_signal_watch()
+		bus.connect('message::element', self.element_message)
+		print(language_file)
+		print(dictionary_file)
 		asr.set_property('lm', language_file)
 		asr.set_property('dict', dictionary_file)
-		asr.set_property('configured', True)
-		#get the Voice Activity DEtectoR
-		self.vad = self.pipeline.get_by_name('vad')
-		self.vad.set_property('auto-threshold',True)
+#		asr.set_property('configured', True)
```

Comment:
I got rid of the segfault by commenting out this line.

You'll see a little debugging code  and some other minor changes that I made after reading 

https://cmusphinx.github.io/wiki/gstreamer/?&#configuring_the_decoder_and_improving_accuracy

But none of that seemed to make a difference until I commented L39 there.